### DISCUSSION TITLE
possibility to use other folders as source from the current one

### DISCUSSION THREAD

**Discussion by RafaelGB** [RafaelGB/obsidian-db-folder#92] [Ideas]
I thinks that add another folders breaks the philosophy and your database source inside the same place it mixes.

**Comment by Trinsec** [comment]
How rigid is this philosophy? And could you elaborate a bit more on the mixing part?

**Comment by RafaelGB** [comment]
sure! the database implies having certain types of objects with similar characteristics in the same place. If we join several databases on the same folder, all those objects (in this case notes), would be mixed.

Add the option of choose another folders it is not very complicated to add and could be another option. If this requirement is very important, it could be raised in the future, but from my point of view it would break that organisation

**Reply by Trinsec** [reply]
Mmm, but what if you have all the same type of objects (the notes) with different views and built-in filters? And you achieve it by having seperate database files for convenience? I've just made another database file in the same folder that I was testing, it seems to work fine? One shows column A, B, C. The other shows column A, D, E. Depending on what I want to look at? 

Example: a movie database. It's one database (one set of objects). But you want to instantly be able to choose 'before 2000' and 'after 2000'. So you make 2 database files out of it. And then another one to show all movies at once. Maybe also show slightly different columns. It's still the same objects, the same sort of data, just shown differently. All the files are appropriately named so it's easy to switch.  The characteristics won't get compromised? Basically all the database files will be similar-ish, just small differences here and there in showing/filtering things, but they're still with the same database-paradigm in mind.

I could make a database file as an extra with all the required columns to see if I didn't miss any data there. But then I'd want to !\[\[\]\]-link one of those database files with only the set of columns I want for viewing.

I've got a game database here, put in folders like 'Games - A' 'Games - B' etc. They're games starting with A, with B, etc. I'd like to have a seperate folder here with a bunch of database files like 'Games - A' etc that'll point to the corresponding folders so I can switch easily between the various game lists I have here.
Each folder also got a 'DLC' folder, I'd also like to be able to toggle right there whether subfolders are shown (thus including the DLCs) or not.

That's kinda how I look at things, a lot of convenience.

By the way, by choosing tags or in/outgoing links as source, isn't that already breaking the organisation since you don't have to put it in any set location either?

(P.s. So far you've made a great plugin here which makes dealing with Dataview a lot easier. :) )

**Comment by RafaelGB** [comment]
Okay, I buy your arguments!
About different kind of filter will be resolver with the same database when this issue arrives https://github.com/RafaelGB/obsidian-db-folder/issues/73

Tags and others sources I think it not breaks the organisation, its another structure.

Your issue will be added to the roadmap. Thank you for your ideas =)

**Reply by ironfroggy** [reply]
Does this mean #73, in addition to the filters it adds, will also add a new folder option to "Source of Database" in the database settings?

**Reply by RafaelGB** [reply]
yes, in addition to that menu, a textarea where you can write another folder path if you choose(not decided name) another_folder

**Comment by lmonson** [comment]
Here's a slightly different take (and an offer to help implement it)

Compare DBFolder to "basic" dataview tables:

A common use case for dataview is to display a table of query results. You can sort that table, clink on links to underlying pages, etc. What you can't do is edit values inside the table or add new rows.

I envision an extension to DBFolder such that:
- DBFolder has a dataview query for populating it's rows.  (Default: "all files from folder X")
- DBFolder has a known folder for storing new files (as it already does).

This would provide the ability pull files from all over your vault for display/edit, while still constraining the displayed columns to those from the .yaml file and/or/intersected with those from the dataview query.

**Reply by RafaelGB** [reply]
hi @lmonson 
Currently exist 4 kind of sources already implemented (current folder, a tag, outgoing and incoming from a link)

The dataview API its the same format of dataviewjs, so if we want to write a query as dataview, we need a mapping from one to another, thats the complexity.

Its inplemented inside helpers/VaultManagement.ts > adapterTfilisToRows function

should be another case option inside sourceDataviewPages function but requires that mapping from string to query dataviewjs with validations and all the stuff
